### DISCUSSION TITLE
self-hosted portainer : cannot connect to default project.

### DISCUSSION THREAD

**Discussion by fundef1** [supabase/supabase#8402] [Questions]
Hi, 
I realize I'm starting on the deep end by running supabase under portainer, but I think I'm almost there... I just can't get studio to connect to the default project.

What I've done so far: 
- followed the self-hosting guide https://supabase.com/docs/guides/hosting/docker
- update .env (update anon/service keys and replace localhost with my dockerhost)
- tweaked the docker-compose to use local volumes for portainer, mapped 7000:8000 8443:7443 (8000 is in use by portainer)
- populated /var/lib/kong with kong.yml (updated the keys too) 
- manually ran all four .sql scripts under docker/db/init straight from the repo
restarted the stack just to make sure

now I don't see any errors any more in the logs... 
but still I can't get past the "Connecting to Default Project" spinner...

any suggestions where to look?
thanks!

**Comment by fundef1** [comment]
Alright... figured it out: the JWT token generator in the self-hosted doc is broken AFAICT.
the original token from the example.env is accepted, ones you generate ..not.
possible iss needs to be supabase-demo instead of supabase.
quick workaround -> generate token -> copy paste into jwt.io -> change iss -> change secret -> copy paste in .env / kong.yml

**Reply by rodoviario** [reply]
I can confirm what you are saying... I was banging my head with de docker compose (and the changed secrets)
After that, I read about the jwt.io and tryied to compare the secrets with the tool on the documentation, and never matched... but never occurred to me that I was going to succeed trying with the jwt.io secrets.
I think that the iss change is not mandatory, but the role is... to "anon" or ""service_role"

**Comment by hf** [comment]
I actually can't seem to find where we pin the `iss` to be `supabase-demo`. AFAICT you have to use [these exact JWTs](https://github.com/supabase/supabase/blob/master/docker/volumes/api/kong.yml#L7-L12) to go through Kong as the `apikey` header.

**Reply by hf** [reply]
Make sure you're using the exact same tokens as generated in `apikey`. Kong does not unpack the JWT to compare contents, it's matching the tokens using string equality.
